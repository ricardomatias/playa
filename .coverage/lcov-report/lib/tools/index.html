<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for lib/tools</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> lib/tools
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.08% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>296/325</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>102/120</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">84.75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>50/59</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">90.76% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>275/303</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="choose.js"><a href="choose.js.html">choose.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="distance.js"><a href="distance.js.html">distance.js</a></td>
	<td data-value="97.37" class="pic high"><div class="chart"><div class="cover-fill" style="width: 97%;"></div><div class="cover-empty" style="width:3%;"></div></div></td>
	<td data-value="97.37" class="pct high">97.37%</td>
	<td data-value="38" class="abs high">37/38</td>
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="18" class="abs high">17/18</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="97.14" class="pct high">97.14%</td>
	<td data-value="35" class="abs high">34/35</td>
	</tr>

<tr>
	<td class="file high" data-value="distribute.js"><a href="distribute.js.html">distribute.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="24" class="abs high">24/24</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	</tr>

<tr>
	<td class="file high" data-value="euclidean.js"><a href="euclidean.js.html">euclidean.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	</tr>

<tr>
	<td class="file high" data-value="friendly.js"><a href="friendly.js.html">friendly.js</a></td>
	<td data-value="97.56" class="pic high"><div class="chart"><div class="cover-fill" style="width: 97%;"></div><div class="cover-empty" style="width:3%;"></div></div></td>
	<td data-value="97.56" class="pct high">97.56%</td>
	<td data-value="82" class="abs high">80/82</td>
	<td data-value="92.59" class="pct high">92.59%</td>
	<td data-value="27" class="abs high">25/27</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	<td data-value="97.4" class="pct high">97.4%</td>
	<td data-value="77" class="abs high">75/77</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="interval.js"><a href="interval.js.html">interval.js</a></td>
	<td data-value="88.89" class="pic high"><div class="chart"><div class="cover-fill" style="width: 88%;"></div><div class="cover-empty" style="width:12%;"></div></div></td>
	<td data-value="88.89" class="pct high">88.89%</td>
	<td data-value="36" class="abs high">32/36</td>
	<td data-value="85" class="pct high">85%</td>
	<td data-value="20" class="abs high">17/20</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="91.43" class="pct high">91.43%</td>
	<td data-value="35" class="abs high">32/35</td>
	</tr>

<tr>
	<td class="file high" data-value="random.js"><a href="random.js.html">random.js</a></td>
	<td data-value="93.33" class="pic high"><div class="chart"><div class="cover-fill" style="width: 93%;"></div><div class="cover-empty" style="width:7%;"></div></div></td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="15" class="abs high">14/15</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="5" class="abs high">4/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="92.86" class="pct high">92.86%</td>
	<td data-value="14" class="abs high">13/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="ring.js"><a href="ring.js.html">ring.js</a></td>
	<td data-value="77.27" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 77%;"></div><div class="cover-empty" style="width:23%;"></div></div></td>
	<td data-value="77.27" class="pct medium">77.27%</td>
	<td data-value="22" class="abs medium">17/22</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="77.27" class="pct medium">77.27%</td>
	<td data-value="22" class="abs medium">17/22</td>
	</tr>

<tr>
	<td class="file high" data-value="roll.js"><a href="roll.js.html">roll.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="rotate.js"><a href="rotate.js.html">rotate.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file high" data-value="shuffle.js"><a href="shuffle.js.html">shuffle.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="time.js"><a href="time.js.html">time.js</a></td>
	<td data-value="71.93" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 71%;"></div><div class="cover-empty" style="width:29%;"></div></div></td>
	<td data-value="71.93" class="pct medium">71.93%</td>
	<td data-value="57" class="abs medium">41/57</td>
	<td data-value="67.86" class="pct medium">67.86%</td>
	<td data-value="28" class="abs medium">19/28</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="56" class="abs medium">40/56</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 20 2019 15:09:45 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
