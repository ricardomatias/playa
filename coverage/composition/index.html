
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for composition</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> composition</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90.49% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>400/442</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.29% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>83/94</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>21/25</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90.49% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>400/442</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="arp.ts"><a href="arp.ts.html">arp.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="42" class="abs high">42/42</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="42" class="abs high">42/42</td>
	</tr>

<tr>
	<td class="file high" data-value="chord-progression.ts"><a href="chord-progression.ts.html">chord-progression.ts</a></td>
	<td data-value="87.69" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 87%"></div><div class="cover-empty" style="width: 13%"></div></div>
	</td>
	<td data-value="87.69" class="pct high">87.69%</td>
	<td data-value="130" class="abs high">114/130</td>
	<td data-value="78.94" class="pct medium">78.94%</td>
	<td data-value="19" class="abs medium">15/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="87.69" class="pct high">87.69%</td>
	<td data-value="130" class="abs high">114/130</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="melodies.ts"><a href="melodies.ts.html">melodies.ts</a></td>
	<td data-value="91.56" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 91%"></div><div class="cover-empty" style="width: 9%"></div></div>
	</td>
	<td data-value="91.56" class="pct high">91.56%</td>
	<td data-value="83" class="abs high">76/83</td>
	<td data-value="94.11" class="pct high">94.11%</td>
	<td data-value="17" class="abs high">16/17</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="91.56" class="pct high">91.56%</td>
	<td data-value="83" class="abs high">76/83</td>
	</tr>

<tr>
	<td class="file high" data-value="motif.ts"><a href="motif.ts.html">motif.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	</tr>

<tr>
	<td class="file high" data-value="percussion.ts"><a href="percussion.ts.html">percussion.ts</a></td>
	<td data-value="96.42" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 96%"></div><div class="cover-empty" style="width: 4%"></div></div>
	</td>
	<td data-value="96.42" class="pct high">96.42%</td>
	<td data-value="56" class="abs high">54/56</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="96.42" class="pct high">96.42%</td>
	<td data-value="56" class="abs high">54/56</td>
	</tr>

<tr>
	<td class="file high" data-value="sequence.ts"><a href="sequence.ts.html">sequence.ts</a></td>
	<td data-value="84.68" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 84%"></div><div class="cover-empty" style="width: 16%"></div></div>
	</td>
	<td data-value="84.68" class="pct high">84.68%</td>
	<td data-value="111" class="abs high">94/111</td>
	<td data-value="90.32" class="pct high">90.32%</td>
	<td data-value="31" class="abs high">28/31</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="16" class="abs medium">12/16</td>
	<td data-value="84.68" class="pct high">84.68%</td>
	<td data-value="111" class="abs high">94/111</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2024-10-16T10:39:20.013Z
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    