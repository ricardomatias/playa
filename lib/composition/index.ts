export { createMotif } from './motif';
export { createChordProgression } from './chord-progression';
export { createMelodies } from './melodies';
export { createPercussion } from './percussion';
export { createArp, genHarmonicShift } from './arp';

export * as Rhythm from './rhythm';
export * from './sequence';
export * from './movement';
