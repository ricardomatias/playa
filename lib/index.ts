// export * from './constants';
// export * from './composition';
// export * from './tools';
// export * from './analysis';

export { Note } from './core/Note';
// export { Chord } from './core/Chord';
export { Scale } from './core/Scale';
export { Key } from './core/Key';
// export * from './core/Time';

// export * from './core/Event';
// export * from './core/ChordEvent';
// export * from './core/NoteEvent';
