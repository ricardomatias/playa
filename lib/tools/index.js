export { default as ring } from './ring';
export { default as choose } from './choose';
export { default as shuffle } from './shuffle';
export { default as distance } from './distance';
export { default as interval } from './interval';
