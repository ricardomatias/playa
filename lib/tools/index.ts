export { choose, chooseMany } from './choose';
export { default as shuffle } from './shuffle';
export { default as rotate } from './rotate';
export { default as distance } from './distance';
export * from './interval';
export { friendly } from './friendly';
export * as Euclidean from './euclidean';
export { default as Sequence } from './sequence';
export { default as Random } from './random';

export * as Rhythm from './rhythm';
export * as Midi from './midi';

export * from './ping';
export { distribute, roll } from '@ricardomatias/roll';

