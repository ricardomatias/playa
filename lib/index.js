import * as Constants from './constants';
import * as Core from './core';
import * as Functional from './functional';
import * as Tools from './tools';

export {
	Core,
	Constants,
	Functional,
	Tools,
};
