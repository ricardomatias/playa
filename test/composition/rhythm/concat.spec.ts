import { Rhythm } from '../../../lib/composition';
import { Random } from '../../../lib/tools';
import '../../matchers';

describe('#concat', () => {
	it('should concat simple', () => {
		// given
		Random.setSeed('test');

		const rhythm = Rhythm.free('2:0:0', Rhythm.Presets.Common);
		const splitRhythm = Rhythm.splitAt(rhythm, '1:0:0');

		// when
		const result = Rhythm.concat(...splitRhythm);

		// then
		expect(result).toLastAround(1920 * 2);
		expect(result).toMatchSnapshot();
	});

	it('should concat 2', () => {
		// given
		Random.setSeed('test');

		const rhythm = Rhythm.free('2:0:0', [ '1nt', '2nd' ]);
		const splitRhythm = Rhythm.splitAt(rhythm, '1:0:0');

		// when
		const result = Rhythm.concat(splitRhythm[0], splitRhythm[1]);

		// then
		expect(result).toLastAround(1920 * 2);
		expect(result).toMatchSnapshot();
	});

	it('should concat 3', () => {
		// given
		Random.setSeed('test');

		const rhy1 = Rhythm.free('1:0:0', [ '4n', '2n' ]);
		const rhy2 = Rhythm.free('1:0:0', [ '4n', '2n' ]);
		const rhy3 = Rhythm.free('1:0:0', [ '4n', '2n' ]);

		// when
		const result = Rhythm.concat(rhy1, rhy2, rhy3);

		// then
		expect(result).toLastAround(1920 * 3);
		expect(result).toMatchSnapshot();
	});
});
