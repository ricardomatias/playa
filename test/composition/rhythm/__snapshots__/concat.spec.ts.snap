// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#concat should concat 2 1`] = `
Array [
  Object {
    "dur": 1280,
    "isRest": false,
    "next": 1280,
    "time": 0,
  },
  Object {
    "dur": 1280,
    "isRest": false,
    "next": 2560,
    "time": 1280,
  },
  Object {
    "dur": 1280,
    "isRest": false,
    "next": 3840,
    "time": 2560,
  },
]
`;

exports[`#concat should concat 3 1`] = `
Array [
  Object {
    "dur": 480,
    "isRest": false,
    "next": 480,
    "time": 0,
  },
  Object {
    "dur": 960,
    "isRest": false,
    "next": 1440,
    "time": 480,
  },
  Object {
    "dur": 480,
    "isRest": false,
    "next": 1920,
    "time": 1440,
  },
  Object {
    "dur": 960,
    "isRest": false,
    "next": 2880,
    "time": 1920,
  },
  Object {
    "dur": 960,
    "isRest": false,
    "next": 3840,
    "time": 2880,
  },
  Object {
    "dur": 480,
    "isRest": false,
    "next": 4320,
    "time": 3840,
  },
  Object {
    "dur": 960,
    "isRest": false,
    "next": 5280,
    "time": 4320,
  },
  Object {
    "dur": 480,
    "isRest": false,
    "next": 5760,
    "time": 5280,
  },
]
`;

exports[`#concat should concat simple 1`] = `
Array [
  Object {
    "dur": 240,
    "isRest": false,
    "next": 240,
    "time": 0,
  },
  Object {
    "dur": 720,
    "isRest": false,
    "next": 960,
    "time": 240,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "next": 1080,
    "time": 960,
  },
  Object {
    "dur": 720,
    "isRest": false,
    "next": 1800,
    "time": 1080,
  },
  Object {
    "dur": 320,
    "isRest": false,
    "next": 2120,
    "time": 1800,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "next": 2240,
    "time": 2120,
  },
  Object {
    "dur": 160,
    "isRest": false,
    "next": 2400,
    "time": 2240,
  },
  Object {
    "dur": 720,
    "isRest": false,
    "next": 3120,
    "time": 2400,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "next": 3240,
    "time": 3120,
  },
  Object {
    "dur": 160,
    "isRest": false,
    "next": 3400,
    "time": 3240,
  },
  Object {
    "dur": 160,
    "isRest": false,
    "next": 3560,
    "time": 3400,
  },
  Object {
    "dur": 160,
    "isRest": false,
    "next": 3720,
    "time": 3560,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "next": 3840,
    "time": 3720,
  },
]
`;
